python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -135,63 +130,68 @@ def makeImage(self):
                                                                    self.config.sourceFlux,
                                                                    self.config.sourceX,
                                                                    self.config.sourceY):
                     if idx == sourceAmp:
-                        self.amplifierAddSource(ampData, sourceFlux, sourceX, sourceY)
+                        self.amplifierAddSource(ampData, sourceFlux * self.config.gain, sourceX, sourceY)
 
             # Other effects in e-
             if self.config.doAddFringe:
-                self.amplifierAddFringe(amp, ampData, np.array(self.config.fringeScale),
+                self.amplifierAddFringe(amp, ampData, np.array(self.config.fringeScale) * self.config.gain,
                                         x0=np.array(self.config.fringeX0),
                                         y0=np.array(self.config.fringeY0))
 
             if self.config.doAddFlat:
                 if ampData.getArray().sum() == 0.0:
+                    # Add noise
                     self.amplifierAddNoise(ampData, 1.0, 0.0)
-                u0 = exposure.getDimensions().getX()
-                v0 = exposure.getDimensions().getY()
+                # Multiply each amplifiers by a Gaussian centered on u0 and v0
+                u0 = exposure.getDetector().getBBox().getDimensions().getX()/2.
+                v0 = exposure.getDetector().getBBox().getDimensions().getY()/2.
                 self.amplifierMultiplyFlat(amp, ampData, self.config.flatDrop, u0=u0, v0=v0)
 
             # ISR effects
-            # 1. Add dark in e- (different from isrMock which does it in ADU)
+            # 1. Add dark in e-
             if self.config.doAddDark:
                 self.amplifierAddNoise(ampData,
                                        self.config.darkRate * self.config.darkTime,
-                                       np.sqrt(self.config.darkRate * self.config.darkTime))
+                                       0. if self.config.calibMode
+                                       else np.sqrt(self.config.darkRate * self.config.darkTime))
 
             # 2. Gain normalize  (from e- to ADU)
             # TODO: DM-43601 gain from PTC per amplifier
             # TODO: DM-36639 gain with temperature dependence
             if self.config.doApplyGain:
                 self.applyGain(ampData, self.config.gain)
 
-            # 3. Add read noise with or without a bias level
-            # to the image region in ADU.
-            self.amplifierAddNoise(ampData, self.config.biasLevel if self.config.doAddBias else 0.0,
-                                   self.config.readNoise / self.config.gain)
+            # 3. Add read noise to the image region in ADU.
+            if not self.config.calibMode:
+                self.amplifierAddNoise(ampData, 0.0,
+                                       self.config.readNoise / self.config.gain)
 
         # 4. Apply cross-talk in ADU
```

Comment:
I believe the crosstalk is defined in electrons (including the new non-linear crosstalk matrices that we'll want to test in the mocks in the near future).  So I think this might be in the wrong place.  (Unless we haven't updated the lsst isr task, in which case we need to do that as well, but maybe a separate ticket but we need that before we cut over to the new task).

Reply:
There is a new units field in the crosstalk code now, and there are gain ratio parameters, but neither of these are currently included in the correction.  I haven't thought about how that should be included, and it depends on where the crosstalk is happening.